clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AbstractAvroStoreClient.java

```diff
@@ -270,6 +275,27 @@ protected RecordSerializer<K> getKeySerializerForRequest() {
         return getKeySerializerWithRetryWithShortInterval();
       } catch (Exception e) {
         whetherStoreInitTriggeredByRequestFail = true;
+        // Kick off an async thread to keep retrying
+        asyncStoreInitThread = new Thread(() -> {
+          while (true) {
+            try {
+              getKeySerializerWithRetryWithShortInterval();
+              whetherStoreInitTriggeredByRequestFail = false;
+              LOGGER.info("Successfully init store client by async store init thread");
+              break;
+            } catch (Exception ee) {
+              LOGGER
+                  .error("Received exception while trying to init store client asynchronously, will keep retrying", ee);
+              try {
+                Thread.sleep(asyncStoreInitSleepIntervalMs);
```

Comment:
Could use `Utils.sleep` (non-throwing flavor) for more compact code.

Reply:
Sure, I can try.
`Utils.sleep` is in `venice-common` and I need to move it to `venice-client-common`

Reply:
It is a little tricky to just move this method into `venice-client-common` since I assume IDEA refactoring doesn't support such, and if I move the whole file to `venice-client-common`, it won't work since `Utils` contains methods depending on other classes defined in `venice-common`.

I am going to copy the sleep implementation into `venice-client-common` and let `Utils.sleep` to call the new one.
Let me know whether it is good enough or not.

Reply:
Sounds good. Thanks